channel.go

```diff
@@ -832,13 +1037,32 @@ channel will be closed.
 
 */
 func (ch *Channel) QueueInspect(name string) (Queue, error) {
+	return ch.QueueInspectContext(context.Background(), name)
+}
+
+/*
+QueueInspectContext passively declares a queue by name to inspect the current
+message count and consumer count.
+
+Use this method to check how many messages ready for delivery reside in the queue,
+how many consumers are receiving deliveries, and whether a queue by this
+name already exists.
+
+If the queue by this name exists, use Channel.QueueDeclare check if it is
+declared with specific parameters.
```

Comment:
With matching parameters. There's a [separate doc section](https://www.rabbitmq.com/queues.html#property-equivalence) that covers this topic.